packages/@ember/-internals/glimmer/lib/syntax/outlet.ts

```diff
@@ -110,7 +110,12 @@ export const outletHelper = internalHelper(
   }
 );
 
-function stateFor(ref: Reference, outlet: OutletState | undefined): OutletDefinitionState | null {
+function stateFor(
+  // SAFETY: default type here is Reference<unknown>, which isn't assignable to the
+  //         type in OutletDefinitionState which is Reference<Outlet | undefiend>
+  ref: Reference<any>,
```

Comment:
This is the wrong fix, consider the input to this function:

```ts
let outletRef = createComputeRef(() => {
  let state = valueForRef(scope.get('outletState') as Reference<OutletState | undefined>);
  return state?.outlets?.main;
});
```

`outletRef` did and should still have type `Reference<OutletState | undefined>`, which appears to be the case since the second argument `outlet` is derived from `valueForRef(ref)`. So the proper fix is to type this argument as `Reference<OutletState | undefined>` and, if that didn't work, trace what broke.

Reply:
idk why this was so complicated initially -- I guess I fixed it later? idk. resolved